File: src/main/java/com/dv/telegram/command/HelpSetting.java
package com.dv.telegram.command;

import com.dv.telegram.WikiBot;
import com.dv.telegram.config.BotSetting;

public class HelpSetting extends BasicBotCommand {

    @Override
    public String getName() {
        return HelpSetting.class.getSimpleName();
    }

    @Override
    public String getDescription(WikiBot bot) {
        return String.format(
            "`%s %s <settingName>` — получить помощь по настройке с названием `<settingName>`.",
            bot.getBotName(),
            getCommandText()
        );
    }

    @Override
    public boolean useMarkdownInResponse() {
        return true;
    }

    @Override
    public String getDefaultCommandName() {
        return "/helpSetting";
    }

    @Override
    public String getResponse(String text, WikiBot bot) {
        int commandStartIndex = text.indexOf(getCommandText());
        if (commandStartIndex < 0) {
            return unknownSettingResponse();
        }

        int commandEndIndex = commandStartIndex + getCommandText().length();
        if (commandEndIndex >= text.length()) {
            return unknownSettingResponse();
        }

        String settingName = text.substring(commandEndIndex).trim();

        BotSetting<?> botSetting = bot.getSettings().getBotSetting(settingName);
        if (botSetting == null) {
            return unknownSettingResponse();
        }

        return String.format("*%s*\n%s", botSetting.getName(), botSetting.getDescription());
    }

    private String unknownSettingResponse() {
        return "Неизвестное имя настройки.";
    }
}

File: src/main/kotlin/com/dv/telegram/command/HelpSetting.kt
package com.dv.telegram.command

import com.dv.telegram.WikiBot
import org.telegram.telegrambots.meta.api.objects.Update

class HelpSetting : BasicBotCommand() {
    override val name: String = javaClass.simpleName

    override fun getDescription(bot: WikiBot) =
        "`${bot.botName} $commandText <settingName>` — получить помощь по настройке с названием `<settingName>`."

    override fun useMarkdownInResponse() = true

    override val defaultCommandName = "/helpSetting"

    override fun getResponse(text: String, bot: WikiBot, update: Update): String {
        val commandStartIndex = text.indexOf(commandText)
        if (commandStartIndex < 0) {
            return unknownSettingResponse()
        }

        val commandEndIndex = commandStartIndex + commandText.length
        if (commandEndIndex >= text.length) {
            return unknownSettingResponse()
        }

        val settingName = text.substring(commandEndIndex).trim()

        val botSetting = bot.settings.getBotSetting(settingName)
            ?: return unknownSettingResponse()

        return "*${botSetting.name}*\n${botSetting.description}"
    }

    private fun unknownSettingResponse() = "Неизвестное имя настройки."
}
